src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj

```diff
@@ -14,6 +14,13 @@
     <PackageDescription>Provides runtime information required to resolve target framework, platform, and runtime specific implementations of .NETCore packages.</PackageDescription>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <!-- When building from source, ensure the RID we're building for is part of the RID graph -->
+    <AdditionalRuntimeIdentifiers>$(AdditionalRuntimeIdentifiers);$(OutputRID)</AdditionalRuntimeIdentifiers>
+    <!-- Also ensure the RID we're bulding on is part of the RID graph, since this may be more specific -->
+    <AdditionalRuntimeIdentifiers>$(AdditionalRuntimeIdentifiers);$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</AdditionalRuntimeIdentifiers>
```

Comment:
I'm not sure we need to add the second.

`OutputRID` should be `$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)` that ignores `DOTNET_RUNTIME_ID` envvar.

Reply:
>`$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)` that ignores `DOTNET_RUNTIME_ID` envvar

`RuntimeInformation.RuntimeIdentifier` does honor the `DOTNET_RUNTIME_ID` envvar.  What did you mean by ignores?

I added both based on your suggestion in the issue.  I think having both might better support a case where someone is building portable Linux from source (OutputRid=linux-x64) and expecting it to support a new distro (myLinux.1.0-x64)

Reply:
>  What did you mean by ignores?

I assume (but maybe this is wrong) that it may be necessary to set `DOTNET_RUNTIME_ID` when building the source-built tarball on a distribution that is not in the graph of the SDK used during the build.
In this case, we'd still want the `OutputRid` to reflect `/etc/os-release`. We'd be setting `DOTNET_RUNTIME_ID` just to make the build work.

Reply:
I see what you mean.  This is a bit of chicken and egg.

How do we even run [the code](https://github.com/dotnet/runtime/blob/3df7a6e3dd4ab11494f4e3a2394b386a73bb446f/src/native/corehost/hostmisc/utils.cpp#L203-L220) to [read from](https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/native/corehost/hostmisc/pal.unix.cpp#L533-L796)`/etc/os-release` to calculate `OutputRID` if that code needs `DOTNET_RUNTIME_ID` set in order to run?  The host needs to "know" what RID it's running on so that it can make decisions about which RID-specific assets to select, and its the same mechanism that impacts [this API](https://github.com/dotnet/runtime/issues/26780).

We could just make the user specify `OutputRID` on the command-line, but that's lame and requires the user to get it right.

If we didn't want that, we'd need someway to run this bit of code from the host and use it during the build, instead of `$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)` as is done today.  This needs to happen outside this task/project since it's something that the build needs to consume (when producing RID-specific packages).  

@agocke @vitek-karas @VSadov what do you think about this: how should we best automatically calculate the "ideal" RID for our purposes of source-build on new platforms.

Reply:
Another thought: should you even need to set `DOTNET_RUNTIME_ID`?  Ideally in the case where a `linux-arch` host build is running on a machine that's say `someDistro.version.arch` it should use `someDistro.version.arch` if that's "known", but if it isn't "known" seems like the host can just assume `linux-arch` for resolving assets (since it's running after all).  Is this how the host works?  Should it?

Reply:
> should you even need to set DOTNET_RUNTIME_ID?

Maybe this will just work, I'm not sure. Let's assume it does.

Once this is merged, we should add a test in source-build repo that verifies this use-case works.
The test can change `/etc/os-release` to something unknown and see if things still build.

Reply:
The host has relatively complicated code to determine the current RID: https://github.com/dotnet/runtime/blob/bc9f00c226c7716c893576d53ea45a616b2b2c36/src/native/corehost/hostmisc/pal.unix.cpp#L710 (there are special versions of this for certain distros, same source file).
This algorithm may "fail", in which case the host falls back to a basic RID - on Linux it will be simply `linux` RID. https://github.com/dotnet/runtime/blob/bc9f00c226c7716c893576d53ea45a616b2b2c36/src/native/corehost/hostmisc/pal.h#L85

If this algorithm works, then there should be no need for `DOTNET_RUNTIME_ID`. If it fails and the fallback is not enough, or if it returns wrong value, then `DOTNET_RUNTIME_ID` is a way to override it and might be necessary. In that case though, we might want to bake the new RID into the hosts produced by the source build somehow, so that running apps using the results of such source build won't require `DOTNET_RUNTIME_ID` to be set. Not sure what the workflow here is...

Reply:
Ah, very cool.  In that case I think not setting `DOTNET_RUNTIME_ID` should just work.  I'll go ahead and remove the second value that's being added to `AdditionalRuntimeIdentifiers`.  I like it, this simplifies the the behavior here.

>Once this is merged, we should add a test in source-build repo that verifies this use-case works.
The test can change /etc/os-release to something unknown and see if things still build.

Can you help with this part @tmds or @omajid?  We probably also want to make sure that the generated runtime.json will include that new RID (to ensure it's flowing through the targets correctly).

Reply:
> Can you help with this part @tmds or @omajid?

@crummel and @MichaelSimons can help with this. I can take a look also, but I'm not familiar with the source-build CI setup.